jpa/eclipselink.jpa.test.jse/src/it/java/org/eclipse/persistence/jpa/test/criteria/TestCriteriaBuilder.java

```diff
@@ -107,4 +111,70 @@ public void testCriteriaCompoundSelectionModel() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testCriteriaBuilder_IN_ClauseLimit() throws Exception {
+        EntityManager em = emf.createEntityManager();
+        try {
+            //"SELECT OBJECT(emp) FROM Employee emp WHERE emp.id IN :result"
+            final CriteriaBuilder builder = em.getCriteriaBuilder();
+            final CriteriaQuery<L1> query = builder.createQuery(L1.class);
+            Root<L1> root = query.from(L1.class);
+            query.where(root.get("name").in(builder.parameter(List.class, "parameterList")));
+
+            Query q = em.createQuery(query);
+
+            //Create a list longer than the limit
+            int limit = getPlatform(emf).getINClauseLimit() + 10;
+            List<String> parameterList = new ArrayList<String>();
+            for(int p = 0; p < limit; p++) {
+                parameterList.add("" + p);
+            }
+            q.setParameter("parameterList", parameterList);
+
+            q.getResultList();
+        } finally {
+            if (em.getTransaction().isActive()) {
+                em.getTransaction().rollback();
+            }
+            if(em.isOpen()) {
+                em.close();
+            }
+        }
+    }
+
+    @Test
+    public void testCriteriaBuilder_NOTIN_ClauseLimit() throws Exception {
```

Comment:
Oracle looks OK, but against MySQL this test fails as You can see from Travis-CI.

Reply:
Interesting! Ill need to revisit this and figure out whats going wrong.